### PR TITLE
Source map update

### PR DESCRIPTION
**Pull request by tomhughes**
This updates things (mostly the tests) to work with newer versions of source-map, up to and including the current 0.5.3 release.

### CODE REVIEW

**Comment by LinusU** [comment]
LGTM from me

File: package.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "test": "mocha"
   },
   "dependencies": {
-    "source-map": "0.1.32"
+    "source-map": "^0.1.33"
```
Comment (julien-f) [inline package.json]: Current version is 0.5.3, why bumping to 0.1.33?

Comment (tomhughes) [inline package.json]: I was just being conservative - that was the first version that needed the changes.

But you're probably right that it should just be bumped all the way up, because a caret version will only match the second digit while the first one is zero.

Comment (julien-f) [inline package.json]: While you are it I suggest trying to update to the latest version :)

Comment (tomhughes) [inline package.json]: I've changed it to 0.5.3 now...

File: test.js [hunk 2]
```diff
@@ -129,7 +139,7 @@ it('normal throw', function() {
     'throw new Error("test");'
   ], [
     'Error: test',
-    /^    at Object\.exports\.test \(.*\/line1\.js:1001:101\)$/
+    /^    at Object\.exports\.test \((?:.*\/)?line1\.js:1001:101\)$/
```
Comment (julien-f) [inline test.js]: Why these changes?

Are they due to the new version of source-map or the version of Node you are using?

Comment (tomhughes) [inline test.js]: It's all down to https://github.com/mozilla/source-map/commit/d281271b106e3e4cc5e06eb8f1589c02cd0d5219 which means that, for example `util.join('.', 'foo.js')` is now `foo.js` rather than `./foo.js` so yes, basically it's down to source-map changing so that it no longer gives us a `./` prefix.

Comment (julien-f) [inline test.js]: IMHO, the tests should be more explicit: do we expect a relative or absolute path?

If relative, we should probably simply remove the `.*/` prefix…

Comment (tomhughes) [inline test.js]: The problem is that `compareStackTrace` runs the test twice, once with a source map in a file on disk and once with an inline source map.

One gives us results with a full path, the other used to give a `./` relative path and now gives a bare name.

The regular expressions need to match both however :-(

Comment (julien-f) [inline test.js]: I see, this is unfortunate because we should test whether our paths are relative or absolute, clients may be relying on this feature…

Maybe `compareStackTrace()` should be updated to be able to specify 2 regexps for a given line if the native and translated traces are different.

@LinusU Any opinion on this?

Comment (LinusU) [inline test.js]: I also think it's a good idea to have stricter tests, so I'm leaning against updating `compareStackTrace` and have 2 regexps.

**Comment by johnjbarton** [comment]
May I suggest moving forward with this update and separately opening an issue to improve the tests?  This package is now quite far behind its most important dependency, source-map.  And time goes by ;-)

**Comment by sindresorhus** [comment]
@LinusU Any chance you could land this?

**Comment by LinusU** [comment]
Sure thing 👍
